src/Integration.Vsix.UnitTests/CFamily/VcxRequestFactoryTests.cs

```diff
@@ -76,5 +317,66 @@ private static Mock<IServiceProvider> CreateServiceProviderReturningProjectItem(
 
             return mockServiceProvider;
         }
+
+        private Request GetSuccessfulRequest(CFamilyAnalyzerOptions analyzerOptions = null,
+            string fileToAnalyze = "c:\\foo\\file.cpp",
+            ICFamilyRulesConfig rulesConfig = null,
+            ProjectItem projectItem = null,
+            Mock<IFileConfig> fileConfig = null,
+            RuleConfigProtocolFormat protocolFormat = null)
+        {
+            rulesConfig ??= Mock.Of<ICFamilyRulesConfig>();
+        
+            var rulesConfigProviderMock = new Mock<ICFamilyRulesConfigProvider>();
+        
+            rulesConfigProviderMock
+                .Setup(x => x.GetRulesConfiguration(It.IsAny<string>()))
+                .Returns(rulesConfig);
+
+            projectItem ??= Mock.Of<ProjectItem>();
+
+            fileConfig ??= CreateDummyFileConfig(fileToAnalyze);
+            var fileConfigProvider = SetupFileConfigProvider(projectItem, analyzerOptions, fileToAnalyze, fileConfig.Object);
+
+            protocolFormat ??= new RuleConfigProtocolFormat("qp", new Dictionary<string, string>());
+
+            var rulesConfigProtocolFormatter = new Mock<IRulesConfigProtocolFormatter>();
+            rulesConfigProtocolFormatter
+                .Setup(x => x.Format(rulesConfig))
+                .Returns(protocolFormat);
+
+            var testSubject = CreateTestSubject(projectItem, 
+                rulesConfigProviderMock.Object, 
+                fileConfigProvider.Object,
+                rulesConfigProtocolFormatter.Object);
+        
+            return testSubject.TryGet(fileToAnalyze, analyzerOptions) as Request;
+        }
+
+        private Mock<IFileConfig> CreateDummyFileConfig(string filePath)
+        {
+            var fileConfig = new Mock<IFileConfig>();
+
+            fileConfig.SetupGet(x => x.PlatformName).Returns("Win32");
```

Comment:
All of these are unfortunately "mandatory" properties in the Captures logic. Empty strings are acceptable but nulls are not...